src/quantity/_src/api.py

```diff
@@ -0,0 +1,96 @@
+"""The Quantity API. Private module."""
+
+__all__ = ["Quantity", "QuantityArray", "Unit"]
+
+from typing import Protocol, runtime_checkable
+
+from astropy.units import UnitBase as Unit
+
+from .array_api import Array
+
+
+@runtime_checkable
+class Quantity(Protocol):
+    """Minimal definition of the Quantity API.
+
+    At minimum a Quantity must have the following attributes:
+
+    - `value`: the numerical value of the quantity (adhering to the Array API)
+    - `unit`: the unit of the quantity
+
+    In practice, Quantities themselves must adhere to the Array API, not just
+    their values. This stricter requirement is described by the `QuantityArray`
+    protocol.
+
+    See Also
+    --------
+    QuantityArray : A Quantity that adheres to the Array API
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> import astropy.units as u
+    >>> from quantity import Quantity
+    >>> from quantity import api
+
+    >>> issubclass(Quantity, api.Quantity)
+    True
+
+    >>> q = Quantity(value=np.array([1, 2, 3]), unit=u.m)
+    >>> isinstance(q, api.Quantity)
+    True
+
+    """
+
+    #: The numerical value of the quantity, adhering to the Array API.
+    value: Array
+
+    #: The unit of the quantity.
+    unit: Unit
+
+    @classmethod
+    def __subclasshook__(cls: type, c: type) -> bool:
+        """Enable the subclass check for non-dataclass descriptors."""
+        return (
+            hasattr(c, "value") or "value" in getattr(c, "__annotations__", ())
```

Comment:
Curious: why the second check? Can there be something in `__annotations__` that is not an attribute?

Reply:
classes don't need to have an `__annotations__` and particularly the value.
In PEP 544 GVR says "isinstance() can be used with both data and non-data protocols, while issubclass() can be used only with non-data protocols. This restriction exists because some data attributes can be set on an instance in constructor and this information is not always available on the class object". This is a way around that for common cases, like property descriptors or dataclasses (via the annotation).

Reply:
We could hold off on adding these methods. Then `api.Quantity` can't be used for `issubclass` checks, only `isinstance` checks...